lib/Backend/Inline.cpp

```diff
@@ -4322,13 +4321,17 @@ Inline::InlineScriptFunction(IR::Instr *callInstr, const FunctionJITTimeInfo *co
     if (callInstr->m_opcode == Js::OpCode::NewScObject || callInstr->m_opcode == Js::OpCode::NewScObjArray)
     {
         isCtor = true;
+
         isFixed = TryOptimizeCallInstrWithFixedMethod(callInstr, inlineeData,
             false /*isPolymorphic*/, false /*isBuiltIn*/, isCtor /*isCtor*/, true /*isInlined*/, safeThis /*&safeThis*/);
-        bool split = SplitConstructorCall(callInstr, true, isFixed, &inlineBailoutChecksBeforeInstr);
-        Assert(split && inlineBailoutChecksBeforeInstr != nullptr);
+
+        inlineBailoutChecksBeforeInstr = PrepareNewScObjArgsForInlining(callInstr);
     }
     else
     {
+        // TODO: should this optimization be tried for NewScObj? The reason it is not is that this
```

Comment:
Is this comment referring to fixed-method optimization? If so, is that what we're going above?